nimble/controller/src/ble_ll_scan.c

```diff
@@ -2413,6 +2414,13 @@ ble_ll_scan_rx_pkt_in(uint8_t ptype, struct os_mbuf *om, struct ble_mbuf_hdr *hd
                init_addr_type = scansm->own_addr_type;
            }
        }
+    } else if (init_addr && ble_ll_resolv_enabled() && ble_ll_is_rpa(init_addr, init_addr_type)) {
+        /* If we are here it means AdvA is identity. Check if initA is RPA */
```

Comment:
if we are here it means adva was not resolved, it does not need to be identity address.
the code below seems ok since it will resolve inita only if adva was in fact identity, but the commit above is confusing (would be ok if put after `if (rl) {`)